lib/thread_local.rb

```diff
@@ -101,17 +101,18 @@ def forget_vanished_threads!
     end
   end
 
-  def define_initializer!
-    ObjectSpace.define_finalizer self do
+  def self.define_initializer!(thread_local, threads, value_key, value_initialized_key)
+    ObjectSpace.define_finalizer thread_local do
```

Comment:
My guess is that this `do ... end` block still capture and then keeps referencing `thread_local` so it is never GCed or finalized.

Do you have any idea we can finalize without capturing `thread_local`?
Instead of using a `do ... end` block, a class method as a callback like described in http://docs.ruby-lang.org/ja/2.0.0/method/ObjectSpace/m/define_finalizer.html ?